Added new blog - Internal issue tracker (#1239)
pankod/refine: documentation/blog/2021-11-12-issue-tracker-refine.md

---

> title: Create Your Easy Customizable Internal Issue Tracker With Refine and Supabase

Reviewer: ```suggestion
title: Create Your Easy Customizable Internal Issue Tracker With refine and Supabase
```

 

> status
* `id` bigint
* `title` varchar

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/zmfn1w8dsqngr5x0apqm.png)

> users
* `email` varchar
* `id` uuid

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/1yu5v8p0a0th5buba34w.png)
 
> tasks
* `id` bigint
* `title` varchar
* `description` varchar
* `start_time` date
* `end_time` date
* `label` bigint
* `priority` bigint
* `statuts` bigint
* `users` uuid

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/jlshwalwk3dd2boz4zx2.png)
 

We created our database tables. The important part here is that as you can see, in our tasks table, label, priority, status and users values ​​are defined as bigint. To summarize the reason for this, we relation the label, priority, status and users tables that we created above with the corresponding values ​​in our tasks table.

 **💡 TIP**: Add Foreign key relationships steps:

1. Got to a table, right click the column you wish to add a foreign key for

2. Select 'edit column'

3. Select 'Add foreign key relation'

4. Select the table and column to make a foreign key for

> 5. Click save

Reviewer: :::tip
Add Foreign key relationships steps:

1. Got to a table, right click the column you wish to add a foreign key for

2. Select 'edit column'

3. Select 'Add foreign key relation'

4. Select the table and column to make a foreign key for

5. Click save
:::

 
`Status Table`

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/p4d8bsrw8jxze2e1mwee.png)
 
 
Let's create a test task to better understand key relation. 

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/mvqfnfsf0sj7qtmnny3f.gif)

As you can see, since we have defined the ids of our label, status, priority and users tables as references to our tasks table, we can now use their values.

Let's look at how we can use this data we created on the Supabase side with refine .

## Refine Project Setup

Now let's refine the task-manager panel. With superplate, we can quickly create a refine project

```bash
  npx superplate-cli refine-task-manager
```

```bash
✔ Select your project type › refine
✔ What will be the name of your app · refine-task-manager
✔ Package manager: · npm
✔ Do you want to customize theme?: · css
✔ Data Provider: · supabase-data-provider
✔ Do you want to customize layout? · custom-layout
✔ i18n - Internationalization: · no
```

After the project setup is loaded, let's start by entering our project and making the necessary changes.

Let's add our supabase url and key to our refine project.

> `src/utility/supabaseClient.ts`
> 
> ```ts

Reviewer: ```suggestion
```ts title="src/utility/supabaseClient.ts"
```

import { createClient } from "@pankod/refine-supabase";

const SUPABASE_URL = "YOUR_SUPABASE_URL";
const SUPABASE_KEY = "YOUR_SUPABASE_API_KEY";

export const supabaseClient = createClient(SUPABASE_URL, SUPABASE_KEY);
```

Now we can access and list the tables we created via the supabase.

- Add custom login and signup page in App.tsx

Our purpose here is to log in if there is a registered user in the supabase. If you do not have a registered user, saving a user to the supabase with refine.
 
#### Custom Login Page

> ```tsx

Reviewer: ```suggestion
```tsx title="src/pages/login/index.tsx"
```

          },
        ] as typeof routerProvider.routes,
      }}
      resources={[
        {
          name: "users",
          list: UserList,
        },
        {
          name: "tasks",
          list: TaskList,
          edit: EditTask,
          create: TaskCreate,
          show: TaskShow,
        },
      ]}
    />
  );
}
```

Our project is done. Lets see how its look like.

`Overview`

<br />

<div class="img-container">
    <div class="window">
        <div class="control red"></div>
        <div class="control orange"></div>
        <div class="control green"></div>
    </div>
    <img src={overview} alt="overview" />
</div>

<br />

`Task Filter`

 ![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/8du6o2q97l6g1773qjwq.gif)

`Dashboard Page`

![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/bphuk3oaufhxacu4xseg.gif)
 

As you can see, we made a simple and short task manager application using refine on our front end and using its data-provider. 

[Here is repo](https://github.com/mlhekinci/refine-supabase-manage-task) 

For more information about Refine: [Refine Github Page](https://github.com/pankod/refine)

> For other examples and articles that will interest you with refine:  [https://dev.to/pankod](https://dev.to/pankod)

Reviewer: ```suggestion
For other examples and articles that will interest you with refine:  [refine blog](https://refine.dev/blog/)
```
